### ISSUE TITLE
CoreFX System.Security.Cryptography namespace

### ISSUE THREAD

**Issue by marek-safar** [mono/mono#9463]
## Types in System.Core.dll to import 

- [ ] AesCng
- [ ] AesCryptoServiceProvider
- [ ] AesManaged
- [ ] CngAlgorithm
- [ ] CngAlgorithmGroup
- [ ] CngExportPolicies
- [ ] CngKey
- [ ] CngKeyBlobFormat
- [ ] CngKeyCreationOptions
- [ ] CngKeyCreationParameters
- [ ] CngKeyHandleOpenOptions
- [ ] CngKeyOpenOptions
- [ ] CngKeyUsages
- [ ] CngProperty
- [ ] CngPropertyCollection
- [ ] CngPropertyOptions
- [ ] CngProvider
- [ ] CngUIPolicy
- [ ] CngUIProtectionLevels
- [ ] ECCurve
- [ ] ECDiffieHellman
- [ ] ECDiffieHellmanPublicKey
- [ ] ECDsa
- [ ] ECDsaCng
- [ ] ECKeyXmlFormat
- [ ] ECParameters
- [ ] ECPoint
- [x] IncrementalHash
- [ ] MD5Cng
- [ ] RSACng
- [ ] SHA1Cng
- [ ] SHA256Cng
- [ ] SHA256CryptoServiceProvider
- [ ] SHA384Cng
- [ ] SHA384CryptoServiceProvider
- [ ] SHA512Cng
- [ ] SHA512CryptoServiceProvider
- [ ] TripleDESCng

**Comment by baulig** [comment]
Some of these overlap with my ongoing X.509 work, so we should coordinate this to avoid possible duplicate work.  From what I can tell so far, I may need to bring the `EC*` types, but I'm not very excited about the rest of the types.

**Comment by CaptainOachkatzl** [comment]
i just want to express my excitement, other than that ignore this comment. :)

**Comment by filipnavara** [comment]
There will likely be some overlap with the X.509 work. The CoreFX classes use the native C System.Security.Cryptography.Native[.Apple] libraries which have to be brought over to Mono. Due to Mono depending on BoringSSL and not OpenSSL it will probably be blocked by other issues (eg. https://github.com/dotnet/corefx/issues/9855 or updating BoringSSL to more recent version).

**Comment by salehmu** [comment]
Does anybody work on those at the current time?

**Comment by filipnavara** [comment]
> Does anybody work on those at the current time?

Nope. We moved on to .NET 6 which already has unified the base class library, including cryptography, across all supported platforms.
